### PR TITLE
[IndexTests] Re-enable `testOutOfDateEvent()` and add a `sleep` call to make it deterministic

### PR DESCRIPTION
**Pull request by akyrtzi**
rdar://87728066

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: Tests/IndexStoreDBTests/IndexTests.swift [hunk 1]
```diff
@@ -317,6 +309,8 @@ final class IndexTests: XCTestCase {
       enableOutOfDateFileWatching: true,
       listenToUnitEvents: true
     )
+    // Wait a bit for the notification to come through.
+    sleep(1)
```
Comment (bnbarham) [inline Tests/IndexStoreDBTests/IndexTests.swift:313]: I assume there's nothing better to wait on here :(?

**Review by bnbarham — APPROVED** [review]

File: Tests/IndexStoreDBTests/IndexTests.swift (hunk 1 above)
Comment (akyrtzi) [reply Tests/IndexStoreDBTests/IndexTests.swift]: yeah, this is junky, I changed it to wait on an XCTest expectation instead.

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: Tests/IndexStoreDBTests/IndexTests.swift [hunk 2]
```diff
@@ -318,6 +324,8 @@ final class IndexTests: XCTestCase {
       listenToUnitEvents: true
     )
 
+    waitForBlock({ delegate.outOfDateInfo != nil })
```
Comment (bnbarham) [inline Tests/IndexStoreDBTests/IndexTests.swift:327]: Much nicer than a sleep :) thanks!

**Review by bnbarham — APPROVED** [review]

**Pull request merged**